addOns/sstiscanner/src/main/java/org/zaproxy/zap/extension/sstiscanner/SSTIScanner.java

```diff
@@ -0,0 +1,501 @@
+
+                    for (SinkPoint sink : sinksToTest) {
+
+                        String output = sink.getCurrentStateInString(newMsg, paramName, renderTest);
+
+                        for (String renderResult : renderExpectedResults) {
+                            // Some rendering tests add html tags so we can not only search for
+                            // the delimiters with the arithmetic result inside. Regex searches
+                            // may be expensive, so first we check if the result exist in the
+                            // response and only then we check if it inside the delimiters and
+                            // was originated by our payload.
+                            String regex =
+                                    "[\\w\\W]*"
+                                            + DELIMITER
+                                            + ".*"
+                                            + renderResult
+                                            + ".*"
+                                            + DELIMITER
+                                            + "[\\w\\W]*";
+
+                            if (output.contains(renderResult) && output.matches(regex)) {
+
+                                String attack =
+                                        Constant.messages.getString(
+                                                MESSAGE_PREFIX + "alert",
+                                                sink.getLocation(),
+                                                output);
+
+                                this.bingo(
+                                        Alert.RISK_HIGH,
+                                        Alert.CONFIDENCE_HIGH,
+                                        getName(),
+                                        getDescription(),
+                                        newMsg.getRequestHeader().getURI().toString(),
+                                        paramName,
+                                        renderTest,
+                                        attack,
+                                        getSolution(),
+                                        newMsg);
+                                found = true;
+                            }
+                        }
+                    }
+                } catch (SocketException ex) {
+                    if (log.isDebugEnabled())
+                        log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage());
+
+                } catch (IOException ex) {
+                    log.warn(
+                            "SSTI vulnerability check failed for parameter ["
+                                    + paramName
+                                    + "]  due to an I/O error",
+                            ex);
+                }
+            }
+        }
+    }
+
+    /** from XXE plugin(yhawke) */
+    private String randomStringFromAlphabet(String alphabet, int length) {
```

Comment:
This helper method seems to exist in multiple classes in this PR, should it be extracted to a Util class?